src/mlpack/methods/ann/models/yolov3/yolov3_layer_impl.hpp

```diff
@@ -0,0 +1,254 @@
+
+  // Could maybe use .each_row()?
+  for (size_t i = 0; i < predictionsPerCell; i++)
+  {
+    anchorsW.col(i).fill(anchors[i * 2]);
+    anchorsH.col(i).fill(anchors[i * 2 + 1]);
+  }
+}
+
+template<typename MatType>
+YOLOv3Layer<MatType>::
+YOLOv3Layer(const YOLOv3Layer& other) :
+    Layer<MatType>(),
+    imgSize(imgSize),
+    numAttributes(numAttributes),
+    gridSize(gridSize),
+    anchorsW(anchorsW),
+    anchorsH(anchorsH),
+    predictionsPerCell(predictionsPerCell)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+YOLOv3Layer<MatType>::
+YOLOv3Layer(YOLOv3Layer&& other) :
+    Layer<MatType>(std::move(other)),
+    imgSize(std::move(imgSize)),
+    numAttributes(std::move(numAttributes)),
+    gridSize(std::move(gridSize)),
+    anchorsW(std::move(anchorsW)),
+    anchorsH(std::move(anchorsH)),
+    predictionsPerCell(std::move(predictionsPerCell))
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+YOLOv3Layer<MatType>&
+YOLOv3Layer<MatType>::
+operator=(const YOLOv3Layer& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(other);
+    imgSize = other.imgSize;
+    numAttributes = other.numAttributes;
+    gridSize = other.gridSize;
+    anchorsW = other.anchorsW;
+    anchorsH = other.anchorsH;
+    predictionsPerCell = other.predictionsPerCell;
+  }
+  return *this;
+}
+
+template<typename MatType>
+YOLOv3Layer<MatType>&
+YOLOv3Layer<MatType>::
+operator=(YOLOv3Layer&& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(std::move(other));
+    imgSize = std::move(other.imgSize);
+    numAttributes = std::move(other.numAttributes);
+    gridSize = std::move(other.gridSize);
+    anchorsW = std::move(other.anchorsW);
+    anchorsH = std::move(other.anchorsH);
+    predictionsPerCell = std::move(other.predictionsPerCell);
+  }
+  return *this;
+}
+
+template <typename MatType>
+void YOLOv3Layer<MatType>::ComputeOutputDimensions()
+{
+  if (this->inputDimensions.size() != 3)
+  {
+    std::ostringstream errMessage;
+    errMessage << "YOLOv3Layer::ComputeOutputDimensions(): "
+               << "Input dimensions must be 3D, but there are "
+               << this->inputDimensions.size() << " input "
+               << "dimensions";
+    throw std::logic_error(errMessage.str());
+  }
+
+  if (this->inputDimensions[0] != this->inputDimensions[1])
+    throw std::logic_error("YOLOv3Layer::ComputeOutputDimensions(): "
+      "Input dimensions must be square.");
+
+  if (gridSize != this->inputDimensions[0] ||
+      gridSize != this->inputDimensions[1])
+  {
+    std::ostringstream errMessage;
+    errMessage << "YOLOv3Layer::ComputeOutputDimensions(): "
+      << "Expected grid size was " << gridSize << " but input dimensions were "
+      << this->inputDimensions[0] << " x " << this->inputDimensions[1];
+    throw std::logic_error(errMessage.str());
+  }
+
+  this->outputDimensions = { numAttributes, grid * predictionsPerCell };
```

Comment:
The output dimensions are primarily for logical interpretation, so should this really be `{ numAttributes, predictionsPerCell, grid }`?  (Are there two grid dimensions to use, too?  I guess maybe it should be `{ numAttributes, predictionsForCell, this->inputDimensions[0], this->inputDimensions[1] }`)

So long as the total number of output dimensions is unchanged, you could change this function without needing to change any of the rest of the implementation.